examples/Model/User.php

```diff
@@ -55,20 +55,11 @@ function apply(array $state, Message ...$events): array
     foreach ($events as $event) {
         switch ($event->messageName()) {
             case UserWasRegistered::class:
-                $state['version'] = 1;
-
-                return array_merge($state, $event->payload(), ['activated' => true]);
+                return array_merge($state, $event->payload(), ['activated' => true, 'version' => 1]);
             case UserWasRegisteredWithDuplicateEmail::class:
-                $state = array_merge($state, $event->payload(), ['activated' => false, 'blocked_reason' => 'duplicate email']);
-                ++$state['version'];
-
-                return $state;
+                return array_merge($state, $event->payload(), ['activated' => false, 'blocked_reason' => 'duplicate email', 'version' => ++$state['version']]);
```

Comment:
Note that we have two distinct methods in the event sourcing aggreate, one for apply and one for replay. Maybe we need this distinction here, too? Or we find a functional way of solving the problem ;)